langchain/vectorstores/deeplake.py

```diff
@@ -293,59 +156,52 @@ def _search_helper(
             List of Documents selected by the specified distance metric,
             if return_score True, return a tuple of (Document, score)
         """
-        view = self.ds
-
-        # attribute based filtering
-        if filter is not None:
-            if isinstance(filter, dict):
-                filter = partial(dp_filter, filter=filter)
-
-            view = view.filter(filter)
-            if len(view) == 0:
-                return []
-
-        if self._embedding_function is None:
-            view = view.filter(lambda x: query in x["text"].data()["value"])
-            scores = [1.0] * len(view)
-
-            if use_maximal_marginal_relevance:
-                raise ValueError(
-                    "For MMR search, you must specify an embedding function on"
-                    "creation."
-                )
-
-        else:
-            emb = embedding or self._embedding_function.embed_query(
-                query
-            )  # type: ignore
-            query_emb = np.array(emb, dtype=np.float32)
-            embeddings = view.embedding.numpy(fetch_chunks=True)
-            k_search = fetch_k if use_maximal_marginal_relevance else k
-            indices, scores = vector_search(
-                query_emb,
+        embedding_function=embedding_function or self._embedding_function
+        
+        if embedding is None:
+            if embedding_function is None:
+                raise ValueError(f"Either `embedding` or `embedding_function` needs to be specified.")
+        
+            embedding = embedding_function.embed_query(query)
+        
+        if isinstance(embedding, list):
+            embedding = np.array(embedding, dtype=np.float32)
+            if len(embedding.shape) > 1:
+                embedding = embedding[0]
+        
+        tensors = self.deepake_vectorstore.search(
```

Comment:
@adolkhan I think we should have a better name for this variable, like `results` or `search_results`. tensors sounds like it could be a list of tensors or tensor names.